lib/spack/spack/binary_distribution.py

```diff
@@ -1063,8 +1070,8 @@ def generate_package_index(cache_prefix, concurrency=32):
         _read_specs_and_push_index(file_list, read_fn, cache_prefix, db, db_root_dir, concurrency)
     except Exception as err:
         msg = "Encountered problem pushing package index to {0}: {1}".format(cache_prefix, err)
-        tty.warn(msg)
-        tty.debug("\n" + traceback.format_exc())
+        tty.debug("\nFailed to push index...\n" + traceback.format_exc())
+        raise GenerateIndexError(msg)
```

Comment:
Get rid of that debug statement, use `raise from`

Reply:
Can you also use `raise SomeError(f"message {variable}")` everywhere?

Reply:
the debug is only there because it was there before, raise from works too.